### ISSUE TITLE
sd3 lora save error

### ISSUE THREAD

**Issue by pzzmyc** [bghira/SimpleTuner#464]
got an unexpected keyword argument 'text_encoder_1_lora_layers_to_save'
![image](https://github.com/bghira/SimpleTuner/assets/43562427/01906935-0d69-433c-98a7-71b542405f4a)

**Comment by bghira** [comment]
are you using `--train_text_encoder`? if so, remove that option. it's not very well-tested, and it'll be a while before I have time to fix it. if you want to open a PR to fix it, I will happily review.

**Comment by pzzmyc** [comment]
> are you using `--train_text_encoder`? if so, remove that option. it's not very well-tested, and it'll be a while before I have time to fix it. if you want to open a PR to fix it, I will happily review.

I attempted to print all the parameters in the main function. Below are the values collected by argparse:
![image](https://github.com/bghira/SimpleTuner/assets/43562427/f7e9a0b6-9432-4287-aba9-5f52cafdbf88)
arg:snr_gamma: 5.0
arg:use_soft_min_snr: False
arg:soft_min_snr_sigma_data: None
arg:model_type: lora
arg:sd3: True
arg:sd3_uses_diffusion: False
arg:weighting_scheme: logit_normal
arg:logit_mean: 0.0
arg:logit_std: 1.0
arg:mode_scale: 1.29
arg:lora_type: None
arg:lora_init_type: default
arg:lora_rank: 16
arg:lora_alpha: 16
arg:lora_dropout: 0.1
arg:controlnet: False
arg:controlnet_model_name_or_path: None
arg:pretrained_model_name_or_path: /media/hhy/A983D940A4049592/ai/SimpleTuner/SimpleTuner/models--stabilityai--stable-diffusion-3-medium-diffusers/snapshots/b1148b4028b9ec56ebd36444c193d56aeff7ab56
arg:pretrained_vae_model_name_or_path: None
arg:prediction_type: epsilon
arg:snr_weight: 1.0
arg:training_scheduler_timestep_spacing: trailing
arg:inference_scheduler_timestep_spacing: trailing
arg:refiner_training: False
arg:refiner_training_invert_schedule: False
arg:refiner_training_strength: 0.2
arg:timestep_bias_strategy: none
arg:timestep_bias_multiplier: 1.0
arg:timestep_bias_begin: 0
arg:timestep_bias_end: 1000
arg:timestep_bias_portion: 0.25
arg:disable_segmented_timestep_sampling: False
arg:rescale_betas_zero_snr: False
arg:vae_dtype: bf16
arg:vae_batch_size: 4
arg:vae_cache_scan_behaviour: recreate
arg:vae_cache_preprocess: False
arg:aspect_bucket_disable_rebuild: False
arg:keep_vae_loaded: False
arg:skip_file_discovery: 
arg:revision: None
arg:variant: None
arg:preserve_data_backend_cache: False
arg:use_dora: False
arg:override_dataset_config: False
arg:cache_dir_text: cache
arg:cache_dir_vae: 
arg:data_backend_config: /media/hhy/A983D940A4049592/ai/SimpleTuner/SimpleTuner/datasets/dataset.json
arg:write_batch_size: 64
arg:enable_multiprocessing: False
arg:aspect_bucket_worker_count: 12
arg:cache_dir: /media/hhy/A983D940A4049592/ai/SimpleTuner/SimpleTuner/datasets/models/cache
arg:cache_clear_validation_prompts: False
arg:caption_strategy: filename
arg:parquet_caption_column: None
arg:parquet_filename_column: None
arg:instance_prompt: None
arg:output_dir: /media/hhy/A983D940A4049592/ai/SimpleTuner/SimpleTuner/datasets/models
arg:seed: 42
arg:seed_for_each_device: True
arg:resolution: 1024.0
arg:resolution_type: pixel
arg:aspect_bucket_rounding: None
arg:aspect_bucket_alignment: 64
arg:minimum_image_size: 1024.0
arg:maximum_image_size: None
arg:target_downsample_size: None
arg:train_text_encoder: False
arg:tokenizer_max_length: None
arg:train_batch_size: 1
arg:num_train_epochs: 10
arg:max_train_steps: None
arg:checkpointing_steps: 500
arg:checkpoints_total_limit: 1
arg:resume_from_checkpoint: latest
arg:gradient_accumulation_steps: 4
arg:gradient_checkpointing: False
arg:learning_rate: 8e-07
arg:text_encoder_lr: None
arg:lr_scale: False
arg:lr_scheduler: sine
arg:lr_warmup_steps: 1000
arg:lr_num_cycles: 1
arg:lr_power: 0.8
arg:use_ema: False
arg:ema_decay: 0.995
arg:non_ema_revision: None
arg:offload_param_path: None
arg:use_8bit_adam: False
arg:use_adafactor_optimizer: False
arg:adafactor_relative_step: False
arg:use_prodigy_optimizer: False
arg:prodigy_beta3: None
arg:prodigy_decouple: True
arg:prodigy_use_bias_correction: True
arg:prodigy_safeguard_warmup: True
arg:prodigy_learning_rate: 0.5
arg:prodigy_weight_decay: 0.01
arg:prodigy_epsilon: 1e-08
arg:use_dadapt_optimizer: False
arg:dadaptation_learning_rate: 1.0
arg:adam_beta1: 0.9
arg:adam_beta2: 0.999
arg:adam_weight_decay: 0.01
arg:adam_epsilon: 1e-08
arg:adam_bfloat16: True
arg:max_grad_norm: 2.0
arg:push_to_hub: False
arg:push_checkpoints_to_hub: False
arg:hub_model_id: None
arg:logging_dir: logs
arg:validation_torch_compile: False
arg:validation_torch_compile_mode: max-autotune
arg:allow_tf32: False
arg:validation_using_datasets: None
arg:webhook_config: None
arg:report_to: wandb
arg:tracker_run_name: simpletuner-sdxl
arg:tracker_project_name: sdxl-training
arg:validation_prompt: 1girl
arg:validation_prompt_library: False
arg:user_prompt_library: None
arg:validation_negative_prompt: blurry, cropped, ugly
arg:num_validation_images: 1
arg:validation_steps: 10000000
arg:num_eval_images: 4
arg:eval_dataset_id: None
arg:validation_num_inference_steps: 30
arg:validation_resolution: 1024
arg:validation_noise_scheduler: euler
arg:validation_disable_unconditional: False
arg:disable_compel: True
arg:enable_watermark: False
arg:mixed_precision: bf16
arg:local_rank: 0
arg:enable_xformers_memory_efficient_attention: False
arg:set_grads_to_none: True
arg:noise_offset: 0.1
arg:noise_offset_probability: 0.25
arg:validation_guidance: 7.5
arg:validation_guidance_rescale: 0.0
arg:validation_randomize: False
arg:validation_seed: 42
arg:fully_unload_text_encoder: False
arg:freeze_encoder_before: 12
arg:freeze_encoder_after: 17
arg:freeze_encoder_strategy: after
arg:freeze_unet_strategy: none
arg:unet_attention_slice: False
arg:print_filenames: False
arg:print_sampler_statistics: False
arg:metadata_update_interval: 65
arg:debug_aspect_buckets: False
arg:debug_dataset_loader: False
arg:freeze_encoder: True
arg:save_text_encoder: False
arg:text_encoder_limit: 25
arg:prepend_instance_prompt: False
arg:only_instance_prompt: False
arg:data_aesthetic_score: 7.0
arg:sdxl_refiner_uses_full_range: False
arg:caption_dropout_probability: 0.1
arg:delete_unwanted_images: False
arg:delete_problematic_images: False
arg:offset_noise: False
arg:lr_end: 4e-7
arg:i_know_what_i_am_doing: False

**Comment by saint-min** [comment]
It's because `SD3LoraLoaderMixin` class in `diffusers/src/diffusers/loaders/lora.py` doesn't store the text encoder lora layers. 
And it looks like this issue has already been fixed! Thanks for your swift response, @bghira.
Just pull the main branch and try again.

**Comment by pzzmyc** [comment]
Thank you very much, it's working fine now. 
![image](https://github.com/bghira/SimpleTuner/assets/43562427/16d13e7a-25db-45da-8b35-327215417cb2)
